app/javascript/components/rooms/RoomJoin.jsx

```diff
@@ -26,28 +31,46 @@ export default function RoomJoin() {
       <div className="vertical-center">
         <FormLogo />
         <Card className="col-md-6 mx-auto p-4 border-0 shadow-sm">
-          <div className="mt-4">
-            <Col>
+          <Row>
+            <Col className="col-8 mt-4">
               <Card.Subtitle className="text-muted mb-1">You have been invited to join</Card.Subtitle>
-              <Card.Title className="pb-2">
+              <h2 className="pt-1">
                 { room.name }
-              </Card.Title>
+              </h2>
             </Col>
-            <Col />
-          </div>
-
-          <Card.Footer className="mt-4 bg-white text-center">
-            <InputGroup className="mt-3">
-              <input
-                type="text"
-                id="join-name"
-                placeholder="Enter your name..."
-                className="form-control"
-                onChange={(event) => setName(event.target.value)}
-              />
-              <Button className="px-4" onClick={() => waitOrJoin(refetchJoin, refetchStatus)}>Join</Button>
-            </InputGroup>
+            <Col className="ms-auto col-4">
+              <Avatar className="float-end" avatar={room.owner.avatar} radius={100} />
+              <p className="float-end">{room.owner.name}</p>
+            </Col>
+          </Row>
 
+          <Card.Footer className="mt-2 bg-white">
+            { waiting ? (
+              <div className="mt-3">
+                <Row>
+                  <Col className="col-10">
+                    <p className="mb-1">The meeting hasn&apos;t started yet</p>
+                    <p className="mb-0 text-muted">You will automatically join when the meeting starts</p>
+                  </Col>
+                  <Col className="col-2">
+                    <Spinner className="float-end" />
+                  </Col>
+                </Row>
+              </div>
+            ) : (
+              <div className="text-center">
+                <InputGroup className="mt-3">
```

Comment:
Is using the controlled form here essential?
I think wherever it's possible `react-hook-form` should be used even for the simple forms for consistency purposes mainly.

Reply:
Using a form here actually overcomplicates things. I prefer we don't use forms unless it's a more than one input